src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,90 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+
+public class Storage {
+    private ArrayList<Task> store;
+    private PrintWriter writer;
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public void storeData(TaskList t) {
+        store = t.getStore();
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < store.size(); i++) {
+            Task curr = store.get(i);
+            switch (curr.getType()) {
+                case TODO:
+                    sb.append("T | " + (curr.isDone ? 1 : 0) + " | " + curr.description + "\n");
+                    break;
+                case EVENT:
+                    sb.append("E | " + (curr.isDone ? 1 : 0) + " | " + curr.description + " | " + curr.time + "\n");
+                    break;
+                case DEADLINE:
+                    sb.append("D | " + (curr.isDone ? 1 : 0) + " | " + curr.description + " | " + curr.time + "\n");
+                    break;
+            }
+        }
+
+        try {
+            writer = new PrintWriter(filePath);
+            writer.write(sb.toString());
+            writer.flush();
+        } catch (FileNotFoundException fnfe) {
+            System.out.println(fnfe);
+        }
+
+    }
+
+    public ArrayList loadData() {
+        store = new ArrayList<>();
+        try {
+            BufferedReader br = new BufferedReader( new FileReader(filePath));
+            String line;
+            while ((line = br.readLine()) != null) {
+                int x = line.indexOf('|');
+                String command = line.substring(0, x-1);
```

Comment:
Operators should be surrounded by a whitespace character (ie. instead of x-1, checkstyle prefers x - 1). This applies to line 66 and 76 as well.